gossip/gossip.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ava-labs/coreth/peer"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+)
+
+type Gossiper[T Tx] interface {
```

Comment:
Trying to figure out if we can get rid of this interface and instead instantiate the request/response from the generic type. I think it's possible but it requires some generics black magic ([ref](https://stackoverflow.com/questions/71791637/cannot-use-variable-of-type-t-as-type-in-argument/71793077#71793077)).